src/wazuh_mcp_server/api/wazuh_client.py

```
@@ -1,568 +1,191 @@
-            # Use error recovery manager for intelligent recovery
-            recovery_result = await error_recovery_manager.handle_error(
-                e,
-                f"wazuh_server_{method.lower()}",
-                retry_func=lambda: self._make_request_internal(method, endpoint, params, data, retry_auth=False),
-                context={"endpoint": endpoint, "method": method, "params": params}
-            )
-            
-            if recovery_result.get("success"):
-                return recovery_result.get("data")
+        except httpx.ConnectError:
+            raise ConnectionError(f"Cannot connect to Wazuh server at {self.config.wazuh_host}:{self.config.wazuh_port}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)